src/main/scala/co/teapot/tempest/server/TempestDBServer.scala

```diff
@@ -75,85 +66,67 @@ class TempestDBServer(databaseClient: TempestDatabaseClient, config: TempestDBSe
   }
 
   def kStepNeighborsFiltered(edgeType: String,
-                             sourceId: Long,
+                             source: Node,
```

Comment:
I noticed that in a number of places you're now accepting as input a Node or IntNode where you were accepting the raw ID before.
But in all these cases, you're then looking up the raw ID from the Node object (eg, `val sourceTempestId = databaseClient.nodeToTempestId(source)`). How about we leave the original method as is, and just write a wrapper around it, that does the conversion?